src/components/Icon/Icon.tsx

```diff
@@ -120,22 +114,22 @@ class Icon extends UIComponent<Extendable<IIconProps>, any> {
     return <ElementType className={classes.root} {...accessibility.attributes.root} {...rest} />
   }
 
-  renderSvgIcon(ElementType, classes, rest, accessibility): React.ReactNode {
+  renderSvgIcon(ElementType, icons: ThemeIcons, classes, rest, accessibility): React.ReactNode {
     const { name } = this.props
-    const icon = name && svgIcons[name]
+    const svgIconDescriptor = icons[name].icon as SvgIconSpec
 
     return (
       <ElementType className={classes.root} {...accessibility.attributes.root} {...rest}>
-        <svg className={classes.svg} viewBox={icon && icon.viewBox}>
-          {icon && icon.element}
-        </svg>
+        {svgIconDescriptor && callable(svgIconDescriptor)({ classes })}
       </ElementType>
     )
   }
 
-  renderComponent({ ElementType, classes, rest, accessibility }) {
-    return this.props.svg
-      ? this.renderSvgIcon(ElementType, classes, rest, accessibility)
+  renderComponent({ ElementType, classes, rest, accessibility, theme }) {
+    const { icons = {} } = theme
+
+    return icons[this.props.name] && icons[this.props.name].isSvg
```

Comment:
I would prefer to access the array only once, store `icons[this.props.name]` to a variable and pass it to `renderSvgIcon()` instead of `icons`